coriolis/tests/api/middleware/test_auth.py

```diff
@@ -0,0 +1,182 @@
+            "mock_user",
+            "mock_tenant",
+            project_name=expected_project_name,
+            project_domain_name=expected_project_domain_name,
+            user_domain_name=expected_user_domain_name,
+            roles=expected_roles,
+            auth_token=expected_auth_token,
+            remote_address=expected_remote_address,
+            service_catalog=expected_service_catalog,
+            request_id=expected_req_id
+        )
+
+        self.assertEqual(
+            req_mock.environ['coriolis.context'],
+            mock_request_context.return_value
+        )
+
+    @mock.patch.object(CoriolisKeystoneContext, "_get_project_id")
+    @mock.patch.object(CoriolisKeystoneContext, "_get_user")
+    def test__call__invalid_service_catalog(
+        self,
+        mock_get_user,
+        mock__get_project_id,
+    ):
+        req_mock = mock.Mock()
+        mock__get_project_id.return_value = 'mock_tenant'
+        mock_get_user.return_value = "mock_user"
+        req_mock.environ.return_value = "mock_environ"
+
+        expected_project_name = 'mock_project_name'
+        expected_project_domain_name = 'mock_project_domain_name'
+        expected_user_domain_name = 'mock_user_domain_name'
+        expected_auth_token = 'mock_token123'
+        expected_service_catalog = {"catalog1": ["service1", "service2"]}
+
+        req_mock.headers = {
+            'X_ROLE': '1,2,3',
+            'X_TENANT_NAME': expected_project_name,
+            'X-Project-Domain-Name': expected_project_domain_name,
+            'X-User-Domain-Name': expected_user_domain_name,
+            'X_AUTH_TOKEN': expected_auth_token,
+            'X_SERVICE_CATALOG':
+                str(expected_service_catalog),
+        }
+
+        self.assertRaises(
+            webob.exc.HTTPInternalServerError,
+            self.auth,
+            req_mock,
+        )
+
+    @mock.patch.object(context, "RequestContext")
+    def test__call__no_headers(self, mock_request_context):
+        req_mock = mock.Mock()
+
+        req_mock.headers = {}
+
+        result = self.auth(req_mock)
+
+        self.assertEqual(
+            webob.exc.HTTPUnauthorized().status,
+            result.status
+        )
+
+    def test_get_project_id_tenant_id(self):
+        req_mock = mock.Mock()
+
+        expected_result = 'mock_tennant'
+
+        req_mock.headers = {
+            'X_TENANT_ID': expected_result,
+        }
+
+        result = self.auth._get_project_id(req_mock)
+
+        self.assertEqual(
+            expected_result,
+            result
+        )
+
+    def test_get_project_id_tenant(self):
+        req_mock = mock.Mock()
+
+        expected_result = 'mock_tennant'
+
+        req_mock.headers = {
+            'X_TENANT': expected_result,
+        }
+
+        result = self.auth._get_project_id(req_mock)
+
+        self.assertEqual(
+            expected_result,
+            result
+        )
+
+    def test_get_project_id_no_tenant(self):
```

Comment:
After doing the changes, this test method should also be updated, check that it raises BadRequest